luacompiler/src/lib/frontend/mod.rs

```diff
@@ -0,0 +1,174 @@
+pub mod constants_map;
+pub mod register_map;
+use self::{constants_map::ConstantsMap, register_map::RegisterMap};
+use bytecode::{
+    instructions::{make_instr, Opcode},
+    LuaBytecode,
+};
+use cfgrammar::RIdx;
+use lrpar::Node::{self, *};
+use lua5_3_l;
+use lua5_3_y;
+use LuaParseTree;
+
+/// Represents a compiler which translates a given Lua parse tree to some bytecode representation.
+/// This compiler will be changed in the future to translate from lua to a higher-level
+/// representation, which is easier to translate to the current bytecode.
+pub struct LuaToBytecode<'a> {
+    pt: &'a LuaParseTree,
+}
+
+impl<'a> LuaToBytecode<'a> {
+    pub fn new(pt: &'a LuaParseTree) -> LuaToBytecode {
+        LuaToBytecode { pt }
+    }
+
+    /// Compile the parse tree to an intermediate representation.
+    pub fn compile_to_ir(&self) -> LuaBytecode {
+        let mut instrs = vec![];
+        let mut pt_nodes: Vec<&Node<u8>> = vec![&self.pt.tree];
```

Comment:
Do you need the static type here?

Reply:
Do you mean the `: Vec<&Node<u8>>`? I think I don't need it anymore.

Reply:
Yes.